File: Question Bank/Not Done/Gol/gol.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, q;
    cin >> n >> q;
    vector<int> a(n), b(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    for (int i = 0; i < n; i++) {
        cin >> b[i];
    }

    for (int i = 0; i < q; i++) {
        int x, y;
        cin >> x >> y;
        a[x - 1] += y;
        int cost = 0;
        for (int j = 1; j < n; j++) {
            if (a[j] < a[j - 1]) {
                int spray = (a[j - 1] - a[j]);
                cost += spray * b[j];
                a[j] += spray;
            }
        }
        cout << cost << endl;
    }

    return 0;
}

File: Question Bank/Not Done/Gol/gol.py
n, q = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

for _ in range(q):
    x, y = map(int, input().split())
    a[x - 1] += y
    cost = 0
    for i in range(1, n):
        if a[i] < a[i - 1]:
            spray = (a[i - 1] - a[i])
            cost += spray * b[i]
            a[i] += spray
    print(cost)